src/shogun/classifier/GaussianProcessClassification.cpp

```diff
@@ -71,18 +71,6 @@ std::shared_ptr<MulticlassLabels> GaussianProcessClassification::apply_multiclas
 	require(m_method->supports_multiclass(), "{} with {} doesn't support "
 			"multi classification\n", m_method->get_name(), lik->get_name());
 
-	// if regression data equals to NULL, then apply classification on training
-	// features
-	if (!data)
-	{
-		if (m_method->get_inference_type()==INF_SPARSE)
```

Comment:
what about these types of checks? that this doesnt support sparse inference methods?